autosubmit/notifications/mail_notifier.py

```diff
@@ -20,59 +20,111 @@
 import smtplib
 import email.utils
 from email.mime.text import MIMEText
+from email.mime.multipart import MIMEMultipart
+from email.mime.application import MIMEApplication
 from log.log import Log
+from pathlib import Path
+from autosubmitconfigparser.config.basicconfig import BasicConfig
+from typing import List, TYPE_CHECKING
+
+
+if TYPE_CHECKING:
+    from autosubmit.platforms.platform import Platform
+
 
 class MailNotifier:
     def __init__(self, basic_config):
         self.config = basic_config
 
-    def notify_experiment_status(self, exp_id,mail_to,platform):
-        message_text = self._generate_message_experiment_status(exp_id, platform)
-        message = MIMEText(message_text)
-        message['From'] = email.utils.formataddr(('Autosubmit', self.config.MAIL_FROM))
-        message['Subject'] = f'[Autosubmit] Warning a remote platform is malfunctioning'
+    def notify_experiment_status(
+            self,
+            exp_id: str,
+            mail_to: str,
+            platform: 'Platform') -> None:
+        message_text = self._generate_message_experiment_status(
+            exp_id, platform)
+        message = MIMEMultipart()
+        message['From'] = email.utils.formataddr(
+            ('Autosubmit', self.config.MAIL_FROM))
+        message['Subject'] = f'[Autosubmit] Warning: a remote platform is malfunctioning'
         message['Date'] = email.utils.formatdate(localtime=True)
+        message.attach(MIMEText(message_text))
+
+        try:
+            files: List[Path] = [p for p in BasicConfig.expid_aslog_dir(
+                exp_id).glob('*_run.log') if p.is_file()]
+            self._attach_files(message, files)
+        except BaseException as e:
```

Comment:
Can be `Exception`, I think? And as we are not using `e`, we don't need to declare it,

```suggestion
        except Exception:
```

Catching `BaseException` is normally a bad idea as it could capture a system error, or when someone presses CTRL+C (in which cases we want the exception to be raised we the program will a) end up in some weird state or b) keep running and hide the error).
I think we can replace the `BaseException` below too.

Reply:
(In reality I think it'd be best to catch the specific error, like `MessageError`, from the mail package, I think... but that's for future, as at the moment in AS we catch Exceptions pretty much everywhere (for another issue).